lib/Data/Validate/Sanctions/Fetcher.pm

```diff
@@ -199,35 +283,51 @@ sub _hmt_csv {
     my $csv = Text::CSV->new({binary => 1}) or die "Cannot use CSV: " . Text::CSV->error_diag();
 
     my @lines = split("\n", $content);
-    my @info;
-    my $i = 0;
-    foreach (@lines) {
-        $i++;
 
-        s/^\s+|\s+$//g;
+    my $parsed = $csv->parse(trim(shift @lines));
+    my @info   = $parsed ? $csv->fields() : ();
+    die 'Publication date was not found' unless @info && _date_to_epoch($info[1]);
 
-        my $status = $csv->parse($_);
-        if (1 == $i) {
-            @info = $status ? $csv->fields() : ();
-            die 'Publish date is invalid' unless @info && _date_to_epoch($info[1]);
-        }
+    my $publish_epoch = _date_to_epoch($info[1]);
+    die 'Publication date is invalid' unless defined $publish_epoch;
+
+    $parsed = $csv->parse(trim(shift @lines));
+    my @row    = $csv->fields();
+    my %column = map { trim($row[$_]) => $_ } (0 .. @row - 1);
+
+    foreach my $line (@lines) {
+        $line = trim($line);
 
-        next unless $status;
+        $parsed = $csv->parse($line);
+        next unless $parsed;
 
         my @row = $csv->fields();
-        my $row = \@row;
-        ($row->[23] and $row->[23] eq "Individual") or next;
-        my $name = _process_name @{$row}[0 .. 5];
 
-        next if $name =~ /^\s*$/;
+        @row = map { trim($_ =~ s/\([^(]*\)$//r) } @row;
```

Comment:
does this erases everything enclosed between parentheses (included) at the end of each row? (wild guess)

Reply:
Yes. They are comments or `country (city)` that's better to be chopped off. I'll add a comment.